src/hooks/useApi.tsx

```diff
@@ -0,0 +1,25 @@
+import { useEffect, useState } from "react";
+import { API } from "../constants";
+
+const useApi = <T,>(route: string) => {
```

Comment:
This is using a generic to derive the type
[Docs](https://www.typescriptlang.org/docs/handbook/2/generics.html
)
Basically we can use useApi<GenericType> to allow hinting for the returned object

You can view on Line 7 that useState has the type Null or T